blogposts/more-intuitive-code-search-on-sourcegraph-dot-com.md

```diff
@@ -0,0 +1,58 @@
+---
+title: "Search the top 10,000 open source repositories on Sourcegraph.com"
+description: "How a hack week project made Sourcegraph.com easier to use, by searching the top 10k open source repositories on GitHub by default."
+author: Issac Trotts, Software Engineer at Sourcegraph
+publishDate: 2019-08-16T00:00-11:55
+tags: [
+  blog
+]
+slug: Search-the-top-10000-open-source-repositories-Sourcegraph-com
+heroImage: /blog/sourcegraph-com-usage.png
+published: true
+---
+
+<p style="text-align: center">
+  <img src="/blog/sourcegraph-com-usage.png" alt="My top feature request is: no new features. Just fix what's broken">
+  <br/><small>Sourcegraph.com stats as of August 2019</small>
+</p>
+
+Wouldn't it be awesome if you could review, browse, and search code from the top ten thousand open source repositories on GitHub? You can, using [Sourcegraph.com](https://sourcegraph.com), and we've now made it easier to use!
+
+## Using Sourcegraph.com like a regular search engine
+
+[Sourcegraph.com](https://sourcegraph.com) indexes over 4 million open source repositories on GitHub! The only problem, was that a [`repo` filter](https://docs.sourcegraph.com/user/search/queries) was required to narrow the repository set to a manageable number, the downside of which, meant you had to know what repositories to search. This meant [Sourcegraph.com](https://sourcegraph.com) was missing out on a key use case⁠—random usage discovery.
+
+Let's say you've recently heard about Google's [Distroless container image](https://github.com/GoogleContainerTools/distroless), and are curious about its usage. Previously, without specifying any repsoitories, your search would've returned this:
+
+![](/blog/too-many-matching-repos.png)
+
+After some [Sourcegraph.com](https://sourcegraph.com) specific code and infrastructure changes, Sourcegraph (by default), will now search the top 10,000 open source repositories from GitHub (ordered by watch count) if a [`repo` filter](https://docs.sourcegraph.com/user/search/queries) was not provided. This means a query for a [distroless based Dockerfile](https://sourcegraph.com/search?q=file:Dockerfile+FROM+gcr.io/distroless) now returns a useful set of results to explore.
```

Comment:
potential neat improvement: link to actually PRs in "specific code and infrastructure changes".

Reply:
Cool idea! I've added links to [#4959](https://github.com/sourcegraph/sourcegraph/pull/4959/files) and [#5062](https://github.com/sourcegraph/sourcegraph/pull/5062/files) as they seem to be the PRs worth highlighting.

@ijt Do I miss PRs?

Reply:
Those are the ones.